packages/aws-cdk/test/integ/cli/common.bash

```diff
@@ -13,7 +13,16 @@ fi
 
 
 if [[ "${STACK_NAME_PREFIX:-}" == "" ]]; then
-  if ${IS_CANARY:-false}; then
+  # Make the stack names unique based on the codebuild project name
+  # (if it exists). This prevents multiple codebuild projects stomping
+  # on each other's stacks and failing them.
+  #
+  # The get codebuild project name from the ID: PROJECT_NAME:1238a83
+  CODEBUILD_PROJECT=$(echo ${CODEBUILD_BUILD_ID:-} | cut -d: -f 1)
+
+  if [[ "${CODEBUILD_PROJECT:-}" != "" ]]; then
+    export STACK_NAME_PREFIX="${CODEBUILD_PROJECT}"
```

Comment:
Could be worth actually having the actual execution ID in there, too, so that even two concurrent executions of the same project don't interfere with each other? I'm not willing to die on that hill though!

Reply:
If we wanted that we could just generate a UUID4 at the start of the run. The downside of that is that we might accumulate garbage, if deploying succeeds but then something fails to clean up the stacks. After a while, our tests will fail because our account has filled up with stale resources, and will need maintenance.

If we do this, we get automatic overwriting of old stacks while still ensuring enough isolation where it matters.